chef/cookbooks/provisioner/recipes/setup_base_images.rb

```diff
@@ -60,8 +63,9 @@
 end
 
 # PXE config
-%w(x86_64).each do |arch|
-  # Make it easy to totally disable/enable ppc64le
+# ppc64le bootloader can parse pxelinux config files
+%w(x86_64 ppc64le).each do |arch|
```

Comment:
Maybe I'm just don't seeing something, but it looks like you filter `discovery_arches` at line 36, and than you basically iterate over same values at lines 67 and 106 (exiting at lines 69 and 108 means you really care about current value of `discovery_arches`)